src/Extractor/QueryFactory.php

```diff
@@ -26,7 +29,16 @@ protected function generate(): string
         $sql[] = 'SELECT ' . $this->getSelect();
         $sql[] = 'FROM ' . $this->getFrom();
 
-        if ($this->config->hasSort()) {
+        if ($this->config->hasIncrementalFetchingKey()) {
+            if (isset($this->inputState['lastFetchedRow'])) {
+                $sql[] = sprintf(
+                    'WHERE %s >= %s',
+                    $this->config->getIncrementalFetchingKey(),
+                    $this->quote($this->inputState['lastFetchedRow'])
+                );
+            }
+            $sql[] = 'ORDER BY ' . $this->config->getIncrementalFetchingKey();
```

Comment:
- `sort` nie je kompatibilny s incremental fetching, ... teraz to ignorujes, bolo by dobre vyhodit chybu v `ConfigDefinition` ...
- `select` tiez nie je kompatibilny, ... lebo ak incremental fetching column nebude sucastou `select`, tak nebude vo vystupe, tiez by som vyhodil chybu